pkg/http/util.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2022 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package http
+
+import (
+	"encoding/base64"
+	"encoding/json"
+
+	"github.com/pkg/errors"
+
+	contribContentType "github.com/dapr/components-contrib/contenttype"
+)
+
+var errContentTypeMismatch = errors.New("error: mismatch between contentType and event")
+
+func convertEventToBytes(event interface{}, contentType string) ([]byte, error) {
+	if contribContentType.IsBinaryContentType(contentType) {
+		// Here the expectation is that for a JSON request, the binary data will be base64 encoded.
+		// When content type is given as binary, try to decode base64 as []byte.
```

Comment:
In general it is that when bin events are to be part of a json request it will ideally be base64 encoded at least that is what we do on normal subscribe . But in this case it will also be documented that bin events need to be base 64 env for bulk publish.